siege_utilities/survey/render.py

```
@@ -0,0 +1,199 @@
+    if chain.table_type == TableType.CROSS_TAB:
+        n_cats = len(chain.views) if chain.views else 0
+        if n_cats > _HEATMAP_THRESHOLD:
+            ct = "heatmap"
+    return ct
+
+
+# ---------------------------------------------------------------------------
+# Figure builders
+# ---------------------------------------------------------------------------
+
+def _build_chart(df: pd.DataFrame, chart_type: str, headline: str) -> Optional[Any]:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_build_chart`

(ANN401)